### PR TITLE
Fix: Library download Problem and unziping

### PR DESCRIPTION
**Pull request by rajendrarox**
Fix Crashing during downloading library

### CODE REVIEW

**Comment by laymintu** [comment]
I tried this version and its worked 

https://github.com/user-attachments/assets/16be1f52-f454-44ae-a5fc-2eb8a656717f

**Comment by nalcwap** [comment]
why this pr not accepted?

**Comment by rajendrarox** [comment]
> why this pr not accepted?

i don't know maybe the maintainer has  busy in other work

**Comment by ilyassesalama** [comment]
I'll review it as soon as I can, but a review from @PranavPurwar or @khaled-0 would be appreciated. I can't recall who made this feature before.

**Comment by rajendrarox** [comment]
> I'll review it as soon as I can, but a review from @PranavPurwar or @khaled-0 would be appreciated. I can't recall who made this feature before.

as your wish my dear friend

File: app/src/main/java/dev/aldi/sayuti/editor/manage/LibraryDownloaderDialogFragment.java [hunk 1]
```diff
@@ -212,7 +213,7 @@ public void onTaskCompleted(@NonNull List<String> dependencies) {
                             var enabledLibs = gson.fromJson(fileContent, Helper.TYPE_MAP_LIST);
                             enabledLibs.addAll(dependencies.stream()
                                     .map(name -> ManageLocalLibraryActivity.createLibraryMap(name, dependencyName))
-                                    .toList());
+                                    .collect(Collectors.toList())); // Replaced toList() with collect(Collectors.toList())
```
Comment (khaled-0) [inline app/src/main/java/dev/aldi/sayuti/editor/manage/LibraryDownloaderDialogFragment.java:216]: No need for comment. code is self explanatory

File: app/src/main/java/mod/pranav/dependency/resolver/DependencyResolver.kt
Comment (khaled-0) [inline app/src/main/java/mod/pranav/dependency/resolver/DependencyResolver.kt:1]: is this necessary? seems irrelevant and unnecessary to me

**Review by khaled-0 — APPROVED** [review]
LGTM

File: app/src/main/java/mod/pranav/dependency/resolver/DependencyResolver.kt [hunk 2]
```diff
@@ -127,10 +127,8 @@ class DependencyResolver(
         open fun invalidPackaging(artifact: Artifact) {}
     }
 
-
     fun resolveDependency(callback: DependencyResolverCallback) {
         eventReciever = callback
-        // this is pretty much the same as `Artifact.downloadArtifact()`, but with some modifications for checks and callbacks
```
Comment (PranavPurwar) [inline app/src/main/java/mod/pranav/dependency/resolver/DependencyResolver.kt:133]: please keep the comments as they are, they make sure future contributors don't make changes that break stuff
